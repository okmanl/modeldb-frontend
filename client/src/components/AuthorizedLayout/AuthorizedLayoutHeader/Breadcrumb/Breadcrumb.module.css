.root {
  font-size: 16px;
  line-height: 19px;
  display: flex;
  align-items: center;
}

.breadcrumb_item {
  display: inline;
}

.arrow {
  margin: 0 9.5px;
  font-size: 14px;
  color: var(--grey-light);
}

.link {
  text-decoration: none;
  color: rgba(255, 255, 255, 0.8);
  outline: none;
  text-transform: uppercase;
}

.link:hover {
  color: var(--bg-color3);
}

.active_link > .link {
  font-weight: 500;
  color: var(--bg-color3);
}
