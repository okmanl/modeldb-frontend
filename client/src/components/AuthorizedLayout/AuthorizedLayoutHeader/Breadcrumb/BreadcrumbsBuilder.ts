import { IRoute } from 'routes';

interface IBreadcrumbs {
  checkLoaded: () => boolean;
  map<T>(f: (item: IBreadcrumbItem<any>) => T): T[];
}

interface IBreadcrumbsBuilder {
  then: <T>(itemSetting: IBreadcrumItemSetting<T>) => IBreadcrumbsBuilder;
  build: (url: string) => IBreadcrumbs;
}

interface IBreadcrumbItem<T> {
  redirectPath: string;
  checkLoaded: () => boolean;
  getName: () => string;
}

interface IBreadcrumItemSetting<T> {
  routes: Array<IRoute<T>>;
  checkLoaded?: (params: T) => boolean;
  getName: (params: T) => string;
}

export default function BreadcrumbsBuilder(
  itemsSetting: Array<IBreadcrumItemSetting<any>> = []
): IBreadcrumbsBuilder {
  return {
    then: itemSetting => BreadcrumbsBuilder(itemsSetting.concat(itemSetting)),
    build: pathname => {
      const breadcrumbs: Array<IBreadcrumbItem<any>> = itemsSetting
        .map(itemSetting => {
          const appropriateRoute = itemSetting.routes.find(({ getMatch }) =>
            Boolean(getMatch(pathname, false))
          )!;

          if (!appropriateRoute) {
            return null;
          }

          const params = appropriateRoute.getMatch(pathname, false);

          const item: IBreadcrumbItem<any> = {
            redirectPath: appropriateRoute.getRedirectPath(params!),
            getName: () => itemSetting.getName(params),
            checkLoaded: () =>
              itemSetting.checkLoaded ? itemSetting.checkLoaded(params) : true,
          };
          return item;
        })
        .filter(Boolean) as Array<IBreadcrumbItem<any>>;

      return makeBreadcrumbs(breadcrumbs);
    },
  };
}

function makeBreadcrumbs(items: Array<IBreadcrumbItem<any>>): IBreadcrumbs {
  return {
    checkLoaded: () => items.every(item => item.checkLoaded()),
    map: f => items.map(item => f(item)),
  };
}
