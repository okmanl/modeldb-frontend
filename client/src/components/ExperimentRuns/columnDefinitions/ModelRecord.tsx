import * as React from 'react';
import { Link } from 'react-router-dom';

import routes from 'routes';

import styles from './ColumnDefs.module.css';

class ModelRecordColDef extends React.Component<any> {
  public render() {
    const { id, projectId, name } = this.props.data;
    return (
      <div className={styles.param_cell}>
        <Link
          className={styles.model_link}
          to={routes.modelRecord.getRedirectPath({
            projectId,
            modelRecordId: id,
          })}
        >
          <div className={styles.modelName_block}>
            <div className={styles.model_name}>
              {name} &nbsp;
              <i className="fa fa-external-link" />
            </div>
          </div>
        </Link>
      </div>
    );
  }
}

export default ModelRecordColDef;
