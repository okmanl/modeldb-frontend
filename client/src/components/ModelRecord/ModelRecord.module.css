.model_layout {
  background-color: var(--bg-color3);
  padding: 18px 30px 30px 30px;
}

.record {
  display: grid;
  grid-template-columns: 175px auto;
  grid-template-rows: 13px auto 12px;
  border-width: 1px 0 0 0;
  border-color: var(--border-color1);
  border-style: solid;
  align-items: top;
}

.record:first-child {
  border: none;
}

.record_divider {
  grid-template-columns: auto;
}

.record_summary {
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-template-rows: 13px auto 12px;
  align-items: top;
}

.tags {
  display: inline-block;
  padding: 5px 3px;
}

.record_summary_main {
  grid-row: 2;
  color: var(--fr-color2);
  line-height: 18px;
}

.record_name {
  font-size: 22px;
  margin-top: 5px;
}

.record_label {
  font-size: 14px;
  color: rgb(141, 141, 141);
}

.meta_block {
  padding: 5px;
  display: flex;
}

.meta_label {
  font-size: 14px;
  color: rgb(141, 141, 141);
}

.meta_value {
  padding-left: 5px;
  color: var(--fr-color2);
}

.record_summary_meta {
  grid-row: 2;
  font-size: 15px;
  line-height: 18px;
}

.record_header {
  grid-row: 2;
  color: var(--fr-color2);
  font-size: 14px;
  font-weight: 500;
  line-height: 18px;
}

.record_header_divider {
  font-size: 18px;
  margin: 5px 0;
}

.record_value {
  grid-row: 2;
  color: var(--fr-color3);
  font-size: 16px;
  line-height: 16px;
}

.record_value.name {
  font-weight: 500;
  color: var(--fr-color2);
}

.record_value.list {
  line-height: 25px;
  margin: -5px 0;
}

.tag {
  height: 34px;
  width: 100px;
  border: 1px solid var(--border-color1);
  border-radius: 2px;
  margin-right: 12px;
  float: left;
  display: grid;
  justify-items: center;
  align-items: center;
}

.tag_text {
  color: #4e4e4e;
  font-size: 13px;
  line-height: 15px;
}

.loader {
  display: flex;
  justify-content: center;
}

.record_serviceBehaviour {
}

.record_dataBehavior {
}

.record_hyperparameters {
}

.record_metrics {
}

.record_artifacts {
}

.record_datasets {
}

.record_deployInfo {
}

.notDeployedMsg {
  display: grid;
  grid-column-gap: 0.5em;
  grid-template-columns: 1em auto;
  font-size: 14px;
  font-style: italic;
  color: var(--text-color3);
  padding: 2em;
  background-color: var(--main-color1-light);
  border: solid 0.5px var(--main-color1);
  border-radius: 5px;
  white-space: pre-wrap;
  max-width: 24em;
  opacity: 0.85;
}
.notDeployedMsg_noMonitoringInfo {
}
.notDeployedMsg_noInfoInTheTimeWindow {
}
