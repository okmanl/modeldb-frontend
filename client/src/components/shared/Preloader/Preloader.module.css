.preloader {
}

.dynamic_size {
  width: 100%;
  height: 100%;
}
