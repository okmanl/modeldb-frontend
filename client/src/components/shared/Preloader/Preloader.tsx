import cn from 'classnames';
import * as React from 'react';

import circlePreloaderSrc from './imgs/circle-preloader.svg';
import dotsPreloaderSrc from './imgs/dots-preloader.gif';
import styles from './Preloader.module.css';

interface ILocalProps {
  variant: 'dots' | 'circle';
  dynamicSize?: boolean;
}

class Preloader extends React.PureComponent<ILocalProps> {
  public render() {
    const { variant, dynamicSize = false } = this.props;
    const preloaderSrc =
      variant === 'dots' ? dotsPreloaderSrc : circlePreloaderSrc;
    return (
      <img
        className={cn(styles.preloader, {
          [styles.dynamic_size]: dynamicSize,
        })}
        src={preloaderSrc}
      />
    );
  }
}

export default Preloader;
