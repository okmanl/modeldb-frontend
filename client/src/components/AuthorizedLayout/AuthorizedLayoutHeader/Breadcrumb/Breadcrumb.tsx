import cn from 'classnames';
import { bind } from 'decko';
import { UnregisterCallback } from 'history';
import { Project } from 'models/Project';
import React from 'react';
import { connect } from 'react-redux';
import { Link, RouteComponentProps, withRouter } from 'react-router-dom';

import Icon from 'components/shared/Icon/Icon';
import Preloader from 'components/shared/Preloader/Preloader';
import ModelRecord from 'models/ModelRecord';
import routes from 'routes';
import { selectExperimentRuns } from 'store/experiment-runs';
import { selectModelRecord } from 'store/model-record';
import { selectProjects } from 'store/projects';
import { IApplicationState, IConnectedReduxProps } from 'store/store';

import styles from './Breadcrumb.module.css';
import BreadcrumbsBuilder from './BreadcrumbsBuilder';

interface IPropsFromState {
  projects?: Project[] | null;
  experimentRuns?: ModelRecord[] | null;
  modelRecord?: ModelRecord | null;
}

interface ILocalState {
  pathname: string;
}

type AllProps = IPropsFromState & RouteComponentProps;

class Breadcrumb extends React.Component<AllProps, ILocalState> {
  public state: ILocalState = {
    pathname: this.props.history.location.pathname.toLowerCase(),
  };

  private unlistenCallback: UnregisterCallback | undefined = undefined;

  public componentDidMount() {
    this.unlistenCallback = this.props.history.listen((location, action) => {
      this.setState({
        ...this.state,
        pathname: location.pathname.toLowerCase(),
      });
    });
  }

  public componentWillUnmount() {
    if (this.unlistenCallback) {
      this.unlistenCallback();
    }
  }

  public render() {
    const breadcrumbs = this.getBreadcrumbs();

    return (
      <div className={styles.root}>
        {!breadcrumbs.checkLoaded() ? (
          <Preloader variant="dots" />
        ) : (
          <>
            {breadcrumbs
              .map(item => (
                <Link className={styles.link} to={item.redirectPath}>
                  {item.getName()}
                </Link>
              ))
              .map((value, index, content) => {
                return index === content.length - 1 ? (
                  <div
                    className={cn(styles.breadcrumb_item, styles.active_link)}
                    key={index}
                  >
                    {value}
                  </div>
                ) : (
                  <React.Fragment key={index}>
                    <div className={styles.breadcrumb_item}>{value}</div>
                    <Icon className={styles.arrow} type="arrow-right" />
                  </React.Fragment>
                );
              })}
          </>
        )}
      </div>
    );
  }

  @bind
  private getBreadcrumbs() {
    const { projects, modelRecord } = this.props;

    return BreadcrumbsBuilder()
      .then({
        routes: [routes.mainPage],
        getName: () => 'projects',
      })
      .then({
        routes: [routes.experimentRuns],
        checkLoaded: params =>
          Boolean(
            projects &&
              projects.some(project => project.id === params.projectId)
          ),
        getName: params =>
          projects!.find(project => project.id === params.projectId)!.name,
      })
      .then({
        routes: [routes.modelRecord],
        checkLoaded: () => Boolean(modelRecord),
        getName: () => modelRecord!.name,
      })
      .build(this.state.pathname);
  }
}

const mapStateToProps = (state: IApplicationState): IPropsFromState => ({
  experimentRuns: selectExperimentRuns(state),
  modelRecord: selectModelRecord(state),
  projects: selectProjects(state),
});

export type IBreadcrumbProps = AllProps;
export { Breadcrumb };
export default withRouter(connect(mapStateToProps)(Breadcrumb));
