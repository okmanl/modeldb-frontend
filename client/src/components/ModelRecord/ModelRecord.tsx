import * as React from 'react';
import { connect } from 'react-redux';
import { RouteComponentProps } from 'react-router';

import Preloader from 'components/shared/Preloader/Preloader';
import tagStyles from 'components/shared/TagBlock/TagBlock.module.css';
import { IArtifact } from 'models/Artifact';
import { IHyperparameter } from 'models/HyperParameters';
import { IMetric } from 'models/Metrics';
import ModelRecord from 'models/ModelRecord';
import routes, { GetRouteParams } from 'routes';
import {
  fetchModelRecord,
  selectIsLoadingModelRecord,
  selectModelRecord,
} from 'store/model-record';
import { IApplicationState } from 'store/store';

import { bindActionCreators, Dispatch } from 'redux';
import styles from './ModelRecord.module.css';
import ShowContentBasedOnUrl from './ShowContentBasedOnUrl/ShowContentBasedOnUrl';

type IUrlProps = GetRouteParams<typeof routes.modelRecord>;

interface IPropsFromState {
  data: ModelRecord | null;
  loadingModelRecord: boolean;
}

interface IActionProps {
  fetchModelRecord: typeof fetchModelRecord;
}

type AllProps = RouteComponentProps<IUrlProps> & IPropsFromState & IActionProps;

class ModelRecordView extends React.PureComponent<AllProps> {
  public componentDidMount() {
    this.props.fetchModelRecord(this.props.match.params.modelRecordId);
  }

  public render() {
    const { data, loadingModelRecord: loading } = this.props;

    return loading ? (
      <div className={styles.loader}>
        <Preloader variant="dots" />
      </div>
    ) : data ? (
      <div className={styles.model_layout}>
        <div className={styles.record_summary}>
          <div className={styles.record_summary_main}>
            <div className={styles.record_label}>Name</div>
            <div className={styles.record_name}>{data.name}</div>
            <br />
            {data.tags && data.tags.length > 0 && (
              <div>
                <div className={styles.record_label}>Tags</div>
                <div>
                  {data.tags.map((value: string, key: number) => {
                    return (
                      <div className={styles.tags} key={key}>
                        <span className={tagStyles.staticTag}>{value}</span>
                      </div>
                    );
                  })}
                </div>
              </div>
            )}
          </div>
          <div className={styles.record_summary_meta}>
            <this.RenderModelMeta label="Id" value={data.id} />
            <this.RenderModelMeta
              label="Experiment"
              value={data.experimentId}
            />
            <this.RenderModelMeta label="Project" value={data.projectId} />
          </div>
        </div>
        {data.codeVersion! && (
          <this.Record
            header="Code version"
            additionalContainerClassName={styles.record_codeVersion}
          >
            {data.codeVersion}
          </this.Record>
        )}
        {data.hyperparameters && data.hyperparameters.length > 0 && (
          <this.Record
            header="Hyperparameters"
            additionalContainerClassName={styles.record_hyperparameters}
          >
            {data.hyperparameters.map((value: IHyperparameter, key: number) => {
              return (
                <div key={key}>
                  <this.RenderModelMeta label={value.key} value={value.value} />
                </div>
              );
            })}
          </this.Record>
        )}
        {data.metrics && data.metrics.length > 0 && (
          <this.Record
            header="Metrics"
            additionalContainerClassName={styles.record_metrics}
          >
            {data.metrics.map((value: IMetric, key: number) => {
              return (
                <div key={key}>
                  <this.RenderModelMeta label={value.key} value={value.value} />
                </div>
              );
            })}
          </this.Record>
        )}
        {data.artifacts && data.artifacts.length > 0 && (
          <this.Record
            header="Artifacts"
            additionalContainerClassName={styles.record_artifacts}
          >
            {data.artifacts.map((value: IArtifact, key: number) => {
              return (
                <div key={key}>
                  <this.RenderModelMeta
                    label={value.key}
                    children={<ShowContentBasedOnUrl path={value.path} />}
                  />
                </div>
              );
            })}
          </this.Record>
        )}
        {data.datasets && data.datasets.length > 0 && (
          <this.Record
            header="Datasets"
            additionalContainerClassName={styles.record_datasets}
          >
            {data.datasets.map((value: IArtifact, key: number) => {
              return (
                <div key={key}>
                  <this.RenderModelMeta
                    label={value.key}
                    children={<ShowContentBasedOnUrl path={value.path} />}
                  />
                </div>
              );
            })}
          </this.Record>
        )}
        )}
      </div>
    ) : (
      ''
    );
  }

  // tslint:disable-next-line:function-name
  private Record(props: {
    header: string;
    children?: React.ReactNode;
    additionalValueClassName?: string;
    additionalContainerClassName?: string;
    additionalHeaderClassName?: string;
  }) {
    const {
      header,
      children,
      additionalValueClassName,
      additionalContainerClassName,
      additionalHeaderClassName,
    } = props;
    return (
      <div className={`${styles.record} ${additionalContainerClassName}`}>
        <div className={`${styles.record_header} ${additionalHeaderClassName}`}>
          {header}
        </div>
        <div className={`${styles.record_value} ${additionalValueClassName}`}>
          {children}
        </div>
      </div>
    );
  }
  // tslint:disable-next-line:function-name
  private RenderModelMeta(props: {
    label: string;
    value?: string | number;
    children?: React.ReactNode;
  }) {
    const { label, value, children } = props;
    return (
      <div className={styles.meta_block}>
        <div className={styles.meta_label}>{`${label} :`}</div>
        {value ? (
          <div className={styles.meta_value}>{value}</div>
        ) : (
          <div>{children}</div>
        )}
      </div>
    );
  }
}

const mapStateToProps = (state: IApplicationState): IPropsFromState => {
  const modelRecord = selectModelRecord(state);
  return {
    data: modelRecord,
    loadingModelRecord: selectIsLoadingModelRecord(state),
  };
};

const mapDispatchToProps = (dispatch: Dispatch): IActionProps => {
  return bindActionCreators(
    {
      fetchModelRecord,
    },
    dispatch
  );
};

export type IModelRecordProps = AllProps;
export { ModelRecordView };
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ModelRecordView);
