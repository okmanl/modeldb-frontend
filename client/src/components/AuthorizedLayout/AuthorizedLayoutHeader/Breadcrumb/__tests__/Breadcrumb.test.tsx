import Preloader from 'components/shared/Preloader/Preloader';
import ModelRecord from 'models/ModelRecord';
import { Project } from 'models/Project';
import routes from 'routes';
import { expRunsMocks } from 'services/mocks/expRunsMock';
import { projectsMock } from 'services/mocks/projectsMock';
import { makeRouterMockProps } from 'utils/tests/mocks/routeComponentProps';
import { makeShallowRenderer } from 'utils/tests/react/helpers';

import { Breadcrumb, IBreadcrumbProps } from '../Breadcrumb';
import styles from '../Breadcrumb.module.css';

const makeShallowComponent = (
  props: Partial<IBreadcrumbProps> = {},
  pathname?: string
) => {
  const routerMockProps = makeRouterMockProps(
    {},
    {},
    {
      history: {
        location: {
          pathname: pathname || routes.mainPage.getRedirectPath({}),
        },
      },
    }
  );
  const defaultProps: IBreadcrumbProps = {
    ...routerMockProps,
    experimentRuns: null,
    modelRecord: null,
    projects: null,
  };
  return makeShallowRenderer<IBreadcrumbProps>(Breadcrumb, defaultProps)(props);
};

const mockProject: Project = {
  id: '123',
  ...projectsMock[0],
} as any;
const mockModelRecord: ModelRecord = {
  id: '123',
  ...expRunsMocks[0],
} as any;

describe('(components/Breadcrumb)', () => {
  it('should render preloader when data for breadcrumbs is not loaded', () => {
    const component = makeShallowComponent(
      {},
      routes.experimentRuns.getRedirectPath({ projectId: mockProject.id })
    );

    expect(component.find(Preloader).length === 1).toBeTruthy();
    expect(component.find(`.${styles.root}`).children.length).toBe(1);
  });

  it('should render "projects" breadcrumb on projects page', () => {
    const component = makeShallowComponent(
      {},
      routes.mainPage.getRedirectPath({})
    );

    expect(component).toMatchSnapshot();
  });

  it('should render "projects > ${project name}" breadcrumbs on experiment runs page', () => {
    const component = makeShallowComponent(
      {
        projects: [mockProject],
      },
      routes.experimentRuns.getRedirectPath({ projectId: mockProject.id })
    );

    expect(component).toMatchSnapshot();
  });

  it('should render "projects > ${project name}" breadcrumbs on charts page', () => {
    const component = makeShallowComponent(
      {
        projects: [mockProject],
      },
      routes.charts.getRedirectPath({ projectId: mockProject.id })
    );

    expect(component).toMatchSnapshot();
  });

  it('should render "projects > ${project name} > ${model record name}" breadcrumbs on charts page', () => {
    const component = makeShallowComponent(
      {
        projects: [mockProject],
        modelRecord: mockModelRecord,
      },
      routes.modelRecord.getRedirectPath({
        projectId: mockProject.id,
        modelRecordId: mockModelRecord.id,
      })
    );

    expect(component).toMatchSnapshot();
  });
});
