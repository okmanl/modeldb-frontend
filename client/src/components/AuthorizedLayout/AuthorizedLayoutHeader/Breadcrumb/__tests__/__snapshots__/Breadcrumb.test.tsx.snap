// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(components/Breadcrumb) should render "projects > \${project name} > \${model record name}" breadcrumbs on charts page 1`] = `
<div
  className="root"
>
  <div
    className="breadcrumb_item"
  >
    <Link
      className="link"
      replace={false}
      to="/"
    >
      projects
    </Link>
  </div>
  <Icon
    className="arrow"
    type="arrow-right"
  />
  <div
    className="breadcrumb_item"
  >
    <Link
      className="link"
      replace={false}
      to="/project/b328df27-f286-4850-8d68-841d2e1a43ba/exp-runs"
    >
      MNIST Multiclassification
    </Link>
  </div>
  <Icon
    className="arrow"
    type="arrow-right"
  />
  <div
    className="breadcrumb_item active_link"
    key="2"
  >
    <Link
      className="link"
      replace={false}
      to="/project/b328df27-f286-4850-8d68-841d2e1a43ba/exp-runs/556c26a1-781b-420f-9343-77cad7c22f9e"
    >
      Run 11
    </Link>
  </div>
</div>
`;

exports[`(components/Breadcrumb) should render "projects > \${project name}" breadcrumbs on charts page 1`] = `
<div
  className="root"
>
  <div
    className="breadcrumb_item"
  >
    <Link
      className="link"
      replace={false}
      to="/"
    >
      projects
    </Link>
  </div>
  <Icon
    className="arrow"
    type="arrow-right"
  />
  <div
    className="breadcrumb_item active_link"
    key="1"
  >
    <Link
      className="link"
      replace={false}
      to="/project/b328df27-f286-4850-8d68-841d2e1a43ba/charts"
    >
      MNIST Multiclassification
    </Link>
  </div>
</div>
`;

exports[`(components/Breadcrumb) should render "projects > \${project name}" breadcrumbs on experiment runs page 1`] = `
<div
  className="root"
>
  <div
    className="breadcrumb_item"
  >
    <Link
      className="link"
      replace={false}
      to="/"
    >
      projects
    </Link>
  </div>
  <Icon
    className="arrow"
    type="arrow-right"
  />
  <div
    className="breadcrumb_item active_link"
    key="1"
  >
    <Link
      className="link"
      replace={false}
      to="/project/b328df27-f286-4850-8d68-841d2e1a43ba/exp-runs"
    >
      MNIST Multiclassification
    </Link>
  </div>
</div>
`;

exports[`(components/Breadcrumb) should render "projects" breadcrumb on projects page 1`] = `
<div
  className="root"
>
  <div
    className="breadcrumb_item active_link"
    key="0"
  >
    <Link
      className="link"
      replace={false}
      to="/"
    >
      projects
    </Link>
  </div>
</div>
`;
